rfcs/0001-restructure.md

```diff
@@ -0,0 +1,285 @@
+
+This would result in the following order groupings in the PHP language family meta-buildpack:
+
+```toml
+[[order]] # HTTPD web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/procfile"
+    version = ""
+    optional = true
+
+[[order]] # Nginx web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/procfile"
+    version = ""
+    optional = true
+
+[[order]] # Built-in web server (web), FPM (php-fpm) & Script (Procfile)
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-builtin-server"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/procfile"
+    version = ""
+    optional = true
+```
+
+## Environment Variables
+
+* `php-dist` buildpack must expose the location of the `php.ini` file through
+  an environment variable available at build and launch.
+* `php-fpm` buildpack must expose the location of the `php-fpm.conf` file through
```

Comment:
@arjun024 I'm unsure that this will be required. I don't see it hurting anything. I just can't think of a reason why another CNB would need to know this. Only the PHP FPM cnb is going to be configuring PHP FPM. 

It's up to you, but I think we could remove this and not cause any problems.

Reply:
This is because `php-nginx` and `php-httpd` has to know this value to set the start command. See the first note in "Unresolved Questions and Bikeshedding".